File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;

/**
 * @author SHIH
 * @date 2019/01/19
 * @description 品牌借口
 **/
public interface BrandService {
    /**
     * 查询所有
     * @return
     */
    public List<TbBrand> getAll();

    /**
     * 分页接口
     * @param pageNum 当前页码
     * @param pageSize 每页数量
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    public void add(TbBrand brand);

    public void update(TbBrand brand);

    public TbBrand findOne(Long id);

    public void delete(Long [] ids);

    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);
}

File: pinyou-goumanager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function ($http) {
    //findall
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }
    this.findPage = function (page,rows) {
        return $http.get("../brand/findPage.do?page="+page+"&size="+rows)
    }
    this.findOne = function (id) {
        return $http.post("../brand/findOne.do?id="+id)
    }
    this.add = function (entity) {
        return $http.post("../brand/add.do",entity)
    }
    this.update = function (entity) {
        return $http.post("../brand/update.do",entity)
    }
    this.dele = function (ids) {
        return $http.post("../brand/delete.do?ids="+ids )
    }
    this.search = function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&size="+rows,searchEntity)
    }
})